modules/git/commit_archive.go

```diff
@@ -20,18 +21,43 @@ const (
 	TARGZ
 )
 
-// CreateArchive create archive content to the target path
-func (c *Commit) CreateArchive(target string, archiveType ArchiveType) error {
-	var format string
-	switch archiveType {
+// String converts an ArchiveType to string
+func (a ArchiveType) String() string {
+	switch a {
 	case ZIP:
-		format = "zip"
+		return "zip"
 	case TARGZ:
-		format = "tar.gz"
-	default:
-		return fmt.Errorf("unknown format: %v", archiveType)
+		return "tar.gz"
 	}
+	return "unknown"
+}
+
+// CreateArchiveOpts represents options for creating an archive
+type CreateArchiveOpts struct {
+	Format ArchiveType
+	Prefix bool
+}
+
+// CreateArchive create archive content to the target path
+func (c *Commit) CreateArchive(target string, opts CreateArchiveOpts) error {
+	if opts.Format.String() == "unknown" {
+		return fmt.Errorf("unknown format: %v", opts.Format)
```

Comment:
@guillep2k this is the line I was referencing.

Reply:
Thanks! I'd missed it.